### ISSUE TITLE
CreateCSharpManifestResourceName.cs vs CreateVisualBasicManifestResourceName.cs

### ISSUE THREAD

**Issue by tmat** [Microsoft/msbuild#2282]
Diffing src\Tasks\CreateCSharpManifestResourceName.cs vs src\Tasks\CreateVisualBasicManifestResourceName.cs shows suspicious differences between the languages. It seems that some bug fixes made to C# has not been applied to VB. 

Ideally, these tasks would share code instead of having separate implementations that diverge over time.

**Comment by tmat** [comment]
cc @nguerrera

**Comment by AndyGerlicher** [comment]
Team Triage:
This sounds great. Marking up for grabs.

**Comment by isiahto** [comment]
Hello,
I have gone through both C# and VB ManifestResourceName class, it seems like they both contain an `internal static string CreateManifestNameImpl` method to generate the same output (according to the unit tests)

Is this issue aimed to extract the impl from the 2 classes?
I would like to work on this issue, please advise!

**Comment by rainersigwald** [comment]
@isiahto Yes, that's the idea. Based on tmat's comments, it sounds like you should start by extracting the C# implementation and then adopt that in the VB one.

We'd love to take a PR that addressed this. Please target the `master` branch, and let us know here if you have any questions. I'll remove the `up-for-grabs` label from this if you want to work on it.

**Comment by isiahto** [comment]
Yes please, thank you.

**Comment by rainersigwald** [comment]
Internally no clear idea whether it diverges intentionally or not; commit message on fix for J# and C# is:

```
   Beta1 bug
    
       CreateJSharpManifestResourceName.java - [309027]
       CreateCSharpManifestResourceName.cs - [309027]
       CreateCSharpManifestResourceName_Tests.cs - [309207]
```

And doesn't say why/why not VB too.

**Comment by BenVillalobos** [comment]
Unassigning myself and adding some context: I recall that VB files had to be considered "special" because of the fact that we manually parse c# files to find the namespace. Talking with Kathleen a while back, I remember it having to do with the differences in namespaces. See `GetFirstClassNameFullyQualified`.

Things may change down the road, but I don't think we can merge these today.

**Comment by YuliiaKovalova** [comment]
Close it as not planned.
